pkg/templatelookup/moduletemplateinfolookup/by_version_strategy.go

```diff
@@ -63,6 +63,10 @@ func (s ByVersionStrategy) filterTemplatesByVersion(ctx context.Context, name, v
 	for _, template := range templateList.Items {
 		if TemplateNameMatch(&template,
 			name) && shared.NoneChannel.Equals(template.Spec.Channel) && template.Spec.Version == version {
+			if template.Spec.Mandatory {
```

Comment:
I doubt we need to put this check in the strategy.

All these strategies can be reused for the mandatory module as well, right? But with your change, they are limited to regular module only.

I would move this filter after the lookup, specifically in the `GetRegularTemplates` [here](https://github.com/kyma-project/lifecycle-manager/blob/fc1d2b80e07144e4e0976a00df441ebc6c96b50e/pkg/templatelookup/regular.go#L72)

Also soon we will basically just keep that by_module_release_meta_strategy, and the rest will be deprecated.